graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DgsDataLoaderProvider.kt

```diff
@@ -45,7 +48,9 @@ import kotlin.system.measureTimeMillis
  */
 class DgsDataLoaderProvider(
     private val applicationContext: ApplicationContext,
-    private val dataLoaderOptionsProvider: DgsDataLoaderOptionsProvider = DefaultDataLoaderOptionsProvider()
+    private val dataLoaderOptionsProvider: DgsDataLoaderOptionsProvider = DefaultDataLoaderOptionsProvider(),
+    private val scheduledExecutorService: ScheduledExecutorService = Executors.newSingleThreadScheduledExecutor(),
```

Comment:
I don't really know when/why to use `singleThreadScheduleExecutor` vs a pool of scheduled threads with `Executors.newScheduledThreadPool(int corePoolSize)`. It might be good to have this configurable, but I'm not sure.

Reply:
Since this is set up as a bean in our autoconfig, I'm thinking the the user can override  the dgsScheduledExecutorService bean to set up differently?

Reply:
I also thought about that and am not sure what a good default is. It probably depends on the load / RPS. If you have a lot of events scheduled at or around the same instant I can see one thread becoming a bottleneck; but I also think it's something that could be solved in a follow up.

Reply:
Thanks for the input folks! The default is just based on what the java-dataloader library already picked. Will merge this PR for now and we can make it configurable if needed in a follow up.

Reply:
Right, but the default uses a new executor for every registry created. That's in and of itself not ideal and runs into its own scaling problem of spinning up too many threads, but it wouldn't run into this specific scaling problem where an executor is shared since it's essentially spinning up a thread per request. I suspect a default like corePoolSize set to the number of CPUs might make more sense. But if the default here is a problem we'll see it in testing, and can fix it in a follow up like you said.